src/main/java/aquality/selenium/browser/LocalBrowserFactory.java

```diff
@@ -2,25 +2,29 @@
 
 import aquality.selenium.configuration.IBrowserProfile;
 import aquality.selenium.configuration.driversettings.IDriverSettings;
+import aquality.selenium.core.localization.ILocalizedLogger;
+import aquality.selenium.core.utilities.IActionRetrier;
 import io.github.bonigarcia.wdm.Architecture;
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.Capabilities;
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.edge.EdgeDriver;
 import org.openqa.selenium.firefox.FirefoxDriver;
 import org.openqa.selenium.ie.InternetExplorerDriver;
 import org.openqa.selenium.remote.RemoteWebDriver;
 import org.openqa.selenium.safari.SafariDriver;
 
-public class LocalBrowserFactory implements BrowserFactory {
+public class LocalBrowserFactory extends BrowserFactory {
 
     private final IBrowserProfile browserProfile;
 
-    public LocalBrowserFactory() {
-        this.browserProfile = AqualityServices.getBrowserProfile();
+    public LocalBrowserFactory(IActionRetrier actionRetrier, IBrowserProfile browserProfile, ILocalizedLogger localizedLogger) {
+        super(actionRetrier, browserProfile, localizedLogger);
+        this.browserProfile = browserProfile;
     }
 
     @Override
-    public Browser getBrowser() {
+    public RemoteWebDriver getDriver() {
```

Comment:
we did not, we return the Browser, but the method is implemented in abstract BF now, because it has the same structure for every factory: create a driver with retry, create a browser with this driver, log that browser is ready.